Documentation/Signoffs/MainProcessor.md

```diff
@@ -0,0 +1,123 @@
+
+# **Function:**
+
+![System](../Images/MainProcessor/mainProcessorOverview.png)
+
+*Figure 1: Main Processor Overview*
+
+The main processor unit is responsible for receiving, analyzing, and interpreting data from the image-detecting system to calculate and aim the interceptor. Its primary function revolves around processing data related to a golf ball's trajectory in real time. The Jetson Nano serves as the brain of the system, consolidating data from the local camera that passes the data to the main processor.
+
+## **Constraints:**
+
+| NO. |	Constraint | Origin |
+|-----|------------|--------|
+| 1	| Time Constraints - Real-time data processing for trajectory prediction for the golf ball can take around 40 ms for each image. The main processor needs to calculate the ball data before the ball reaches the end which varies from 1.9 seconds to 7.4 seconds | System Constraint |
+| 2	| Processing Speed - The main processor scripts and programs have to be optimized for efficient calculations. The scripts get the speed, wire, and variable height from input data. These calculations should not take longer than 40 ms per calculation iteration | System Constraint |
+| 3	| Signal Interpretation Challenges - Ensuring accurate data interpretation that is received from the Arduino's needs to be 5 Volts otherwise the main processor will not accept the data and could react inappropriately | Programming Constraint |
+| 4	| Resource Utilization - Since the board has 1.43GHz with quadcores and 4GB RAM, the main processor needs to be utilized properly to prevent an overload of system resources. The system needs to use all cores and not overload the RAM for speed efficiency but not sacrifice stability | System Constraint |
+| 5 | Pausing Processes: The system needs a pause state to stop other scripts from activating firing mechanisms. | Rulebook |
+
+
+
+## **Fulfilling Constraints:**
+
+**Time Constraints:** Real-time processing of sensor data and trajectory calculations impose time constraints on the Jetson Nano. Since it has to be able to detect and calculate the proper position of the ball. Its 1.43GHz quad-core ARM Cortex-A57 the processor needs to be able to receive, process, calculate the interceptor’s path, and aim the interceptor before the golf ball gets too far down the string [3]. THe ball's travel time varies from 1.9 seconds to 7.4 seconds and the programs and data transmission needs to be optimized for an accurate and efficient system to be able to run fast enough. Delays in data acquisition, processing, or interceptor firing may affect the interception accuracy dramatically [2].
```

Comment:
What is the limit? Assuming worst case scenario, how long does the processor have to do the computations? How do you know?